[hiredis[ssl]] build failure

**Host Environment**
- OS: Windows 64 bit
- Compiler: VS2019 Version 16.6.2

**To Reproduce**
Steps to reproduce the behavior:
vcpkg upgrade hiredis[ssl]:x64-windows-static
vcpkg upgrade hiredis[ssl]:x64-windows-static --no-dry-run
vcpkg install hiredis[ssl]:x64-windows-static --recurse
whereas vcpkg install hiredis[core]:x64-windows-static is building successfully

**Failure logs** 

**\vcpkg\buildtrees\hiredis\src\4889033d94-d35835c897.clean\ssl.c(37): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory**

Computing installation plan...
The following packages will be rebuilt:
    hiredis[core,ssl]:x64-windows-static
Starting package 1/2: hiredis:x64-windows-static
Removing package hiredis:x64-windows-static...
Removing package hiredis:x64-windows-static... done
Elapsed time for package hiredis:x64-windows-static: 83.49 ms
Starting package 2/2: hiredis:x64-windows-static
Building package hiredis[core,ssl]:x64-windows-static...
-- Using cached  /vcpkg/downloads/redis-hiredis-e777b0295eeeda89ee2ecef6ec5cb54889033d94.tar.gz
-- Cleaning sources at  /vcpkg/buildtrees/hiredis/src/4889033d94-d35835c897.clean. Pass --editable to vcpkg to reuse sources.
-- Extracting source  /vcpkg/downloads/redis-hiredis-e777b0295eeeda89ee2ecef6ec5cb54889033d94.tar.gz
-- Applying patch fix-feature-example.patch
-- Applying patch support-static-in-win.patch
-- Applying patch fix-timeval.patch
-- Using source at  /vcpkg/buildtrees/hiredis/src/4889033d94-d35835c897.clean
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed:  /vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory:  /vcpkg/buildtrees/hiredis/x64-windows-static-dbg
    See logs for more information:
      \vcpkg\buildtrees\hiredis\install-x64-windows-static-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/hiredis/portfile.cmake:28 (vcpkg_install_cmake)
  scripts/ports.cmake:76 (include)


Error: Building package hiredis:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: hiredis:x64-windows-static
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.

--- a/ports/hiredis/CONTROL
+++ b/ports/hiredis/CONTROL
@@ -10,4 +10,4 @@ Build-Depends: openssl
 
 Feature: example
 Description: Build example
-Build-Depends: libevent, pthread
+Build-Depends: libevent, pthread, libuv

--- a/ports/hiredis/CONTROL
+++ b/ports/hiredis/CONTROL
@@ -1,5 +1,6 @@
 Source: hiredis
-Version: 2019-11-2-1
+Version: 2019-11-2
+Port-Version: 2
 Homepage: https://github.com/redis/hiredis
 Description: Hiredis is a minimalistic C client library for the Redis database.
 

--- a/ports/hiredis/fix-feature-example.patch
+++ b/ports/hiredis/fix-feature-example.patch
@@ -1,8 +1,8 @@
 diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
-index dd3a313..0df75d5 100644
+index dd3a313..8c69d3a 100644
 --- a/examples/CMakeLists.txt
 +++ b/examples/CMakeLists.txt
-@@ -19,10 +19,17 @@ if (LIBEV)
+@@ -19,16 +19,30 @@ if (LIBEV)
      TARGET_LINK_LIBRARIES(example-libev hiredis ev)
  ENDIF()
  
@@ -22,8 +22,22 @@ index dd3a313..0df75d5 100644
 +    TARGET_INCLUDE_DIRECTORIES(example-libevent PRIVATE ${LIBEVENT_INCLUDES})
  ENDIF()
  
++FIND_LIBRARY(UV_LIBRARY libuv)
  FIND_PATH(LIBUV uv.h)
-@@ -38,9 +45,17 @@ IF (APPLE)
+ IF (LIBUV)
+     ADD_EXECUTABLE(example-libuv example-libuv.c)
+-    TARGET_LINK_LIBRARIES(example-libuv hiredis uv)
++    if(WIN32)
++        set(LIB_LISTS Iphlpapi.lib Psapi.lib Userenv.lib)
++    else()
++        set(LIB_LISTS)
++    endif()
++    TARGET_LINK_LIBRARIES(example-libuv hiredis ${UV_LIBRARY} ${LIB_LISTS})
++    TARGET_INCLUDE_DIRECTORIES(example-libuv PRIVATE ${LIBUV})
+ ENDIF()
+ 
+ IF (APPLE)
+@@ -38,9 +52,21 @@ IF (APPLE)
  ENDIF()
  
  IF (ENABLE_SSL)
@@ -37,13 +51,30 @@ index dd3a313..0df75d5 100644
 +    ENDIF()
      ADD_EXECUTABLE(example-ssl example-ssl.c)
 -    TARGET_LINK_LIBRARIES(example-ssl hiredis hiredis_ssl)
-+    TARGET_LINK_LIBRARIES(example-ssl hiredis hiredis_ssl OpenSSL::SSL OpenSSL::Crypto ${THREADS_LIBS})
++    if(WIN32)
++        TARGET_LINK_LIBRARIES(example-ssl hiredis hiredis_ssl OpenSSL::SSL OpenSSL::Crypto ${THREADS_LIBS} crypt32.lib)
++    else()
++        TARGET_LINK_LIBRARIES(example-ssl hiredis hiredis_ssl OpenSSL::SSL OpenSSL::Crypto ${THREADS_LIBS})    
++    endif()
  ENDIF()
  
  ADD_EXECUTABLE(example example.c)
 -TARGET_LINK_LIBRARIES(example hiredis)
 +TARGET_LINK_LIBRARIES(example hiredis ${LIBEVENT_LIBS})
-\ No newline at end of file
+diff --git a/examples/example-libuv.c b/examples/example-libuv.c
+index a5462d4..9b7ca3e 100644
+--- a/examples/example-libuv.c
++++ b/examples/example-libuv.c
+@@ -33,7 +33,9 @@ void disconnectCallback(const redisAsyncContext *c, int status) {
+ }
+ 
+ int main (int argc, char **argv) {
++#ifndef _WIN32
+     signal(SIGPIPE, SIG_IGN);
++#endif
+     uv_loop_t* loop = uv_default_loop();
+ 
+     redisAsyncContext *c = redisAsyncConnect("127.0.0.1", 6379);
 diff --git a/examples/example-ssl.c b/examples/example-ssl.c
 index 81f4648..9f42923 100644
 --- a/examples/example-ssl.c

--- a/ports/hiredis/fix-pthread.h-not-found-on-windows.patch
+++ b/ports/hiredis/fix-pthread.h-not-found-on-windows.patch
@@ -1,0 +1,16 @@
+diff --git a/ssl.c b/ssl.c
+index 78ab9e4..768f304 100644
+--- a/ssl.c
++++ b/ssl.c
+@@ -34,7 +34,11 @@
+ #include "async.h"
+ 
+ #include <assert.h>
++#ifdef _WIN32
++#include <windows.h>
++#else
+ #include <pthread.h>
++#endif
+ #include <errno.h>
+ #include <string.h>
+ 

--- a/ports/hiredis/portfile.cmake
+++ b/ports/hiredis/portfile.cmake
@@ -12,6 +12,7 @@ vcpkg_from_github(
         fix-feature-example.patch
         support-static-in-win.patch
         fix-timeval.patch
+        fix-pthread.h-not-found-on-windows.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

